08_cell_line_prediction/nn_diagnostics/nbconverted/nn_dropout_range.py

```diff
@@ -0,0 +1,158 @@
+
+import pancancer_evaluation.config as cfg
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+results_dir = os.path.join(
+    cfg.repo_root, '08_cell_line_prediction', 'results', 'tcga_ccle_nn_dropout_range'
+)
+
+num_genes = 8000
+plot_gene = 'KRAS'
+
+
+# In[3]:
+
+
+dropout_dfs = {}
+
+for dropout_file in glob.glob(
+    os.path.join(results_dir,
+                 plot_gene,
+                 f'{plot_gene}_signal_mad_s42_n{num_genes}_d*_classify_learning_curves.tsv.gz')
+):
+    dropout = (
+        os.path.basename(dropout_file).split('_')[5].replace('d', '')
+    )
+    dropout_df = pd.read_csv(dropout_file, sep='\t', index_col=0).reset_index(drop=True)
+    dropout_df['dropout'] = dropout
+    dropout_dfs[dropout] = dropout_df
+
+print(sorted(list(dropout_dfs.keys())))
+dropout_dfs[list(dropout_dfs.keys())[0]].head()
+
+
+# In[4]:
+
+
+sns.set_style('whitegrid')
+sns.set({'figure.figsize': (24, 10)})
+
+dropout_vals = sorted(list([float(k) for k in dropout_dfs.keys()]))
+print(dropout_vals)
+
+fig, axarr = plt.subplots(ceil(len(dropout_vals) / 3), 3)
+
+for ix, dropout in enumerate(dropout_vals):
+    ax = axarr[ix // 3, ix % 3]
+    if ix == 0:
+        sns.lineplot(data=dropout_dfs[str(dropout)], x='epoch', y='value', hue='dataset', ax=ax)
+    else:
+        sns.lineplot(data=dropout_dfs[str(dropout)], x='epoch', y='value', hue='dataset', legend=False, ax=ax)
+    ax.set_xlabel('Epoch')
+    ax.set_ylabel('AUPR')
+    ax.set_ylim(0, 1.1)
+    ax.set_title(f'Learning curve for {plot_gene}, dropout={dropout}, averaged over splits')
+    
+plt.tight_layout()
+
+
+# In[5]:
+
+
+sns.set_style('whitegrid')
+sns.set({'figure.figsize': (24, 10)})
+
+fig, axarr = plt.subplots(ceil(len(dropout_vals) / 3), 3)
+
+for dropout_ix, dropout in enumerate(dropout_vals):
+    ax = axarr[dropout_ix // 3, dropout_ix % 3]
+    dropout_df = dropout_dfs[str(dropout)]
+    for fold_ix, fold in enumerate(dropout_df.fold.unique()):
+        if dropout_ix == 0 and fold_ix == 0:
+            sns.lineplot(data=dropout_df[dropout_df.fold == fold],
+                         x='epoch', y='value', hue='dataset', ax=ax)
+        else:
+            sns.lineplot(data=dropout_df[dropout_df.fold == fold],
+                         x='epoch', y='value', hue='dataset',
+                         legend=False, ax=ax)
+    ax.set_xlabel('Epoch')
+    ax.set_ylabel('AUPR')
+    ax.set_ylim(0, 1.1)
+    ax.set_title(f'Learning curve for {plot_gene}, dropout={dropout}, for each split')
+
+plt.tight_layout()
+
+
+# In[6]:
+
+
+dropout_df = pd.concat(dropout_dfs.values())
+print(dropout_df.shape)
+dropout_df.head()
+
+
+# In[7]:
+
+
+last_epoch = dropout_df.epoch.max()
+perf_df = (dropout_df[dropout_df.epoch == last_epoch]
+    .drop(columns=['epoch'])
+)
+                      
+print(perf_df.shape)
+perf_df.head()
+
+
+# In[8]:
+
+
+# same plot as before but with the "best"/"smallest" parameters marked
```

Comment:
It doesn't look like the "best"/"smallest" parameters are marked here, is the comment outdated?